### ISSUE TITLE
Build failure with GHC 9.6 (base-4.18)

### ISSUE THREAD

**Issue by andreasabel** [Lysxia/generic-data#60]
Building `generic-data-1.0.0.0` with GHC 9.6.1 alpha2 produces this error and few similar ones:
```
src/Generic/Data/Internal/Generically.hs:250:10: error: [GHC-43085]
    • Overlapping instances for Eq (Generically1 f a)
        arising from a use of ‘ghc-prim-0.10.0:GHC.Classes.$dm/=’
      Matching instances:
        instance forall k (f :: k -> *) (a :: k).
                 (Generic1 f, Eq (Rep1 f a)) =>
                 Eq (Generically1 f a)
          -- Defined in ‘GHC.Generics’
        instance (Generic1 f, Eq1 (Rep1 f), Eq a) => Eq (Generically1 f a)
          -- Defined at src/Generic/Data/Internal/Generically.hs:250:10
    • In the expression:
        ghc-prim-0.10.0:GHC.Classes.$dm/= @(Generically1 f a)
      In an equation for ‘/=’:
          (/=) = ghc-prim-0.10.0:GHC.Classes.$dm/= @(Generically1 f a)
      In the instance declaration for ‘Eq (Generically1 f a)’
    |
250 | instance (Generic1 f, Eq1 (Rep1 f), Eq a) => Eq (Generically1 f a) where
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by Lysxia** [comment]
Thanks for the report. I released 1.0.0.1 with the fix.

**Comment by andreasabel** [comment]
Excellent, thanks!
